plugins/building-hacks.cpp

```diff
@@ -289,15 +292,22 @@ struct work_hook : df::building_workshopst{
             std::vector<graphic_tile> &cur_frame=def->frames[frame];
             for(size_t i=0;i<cur_frame.size();i++)
             {
-                if(cur_frame[i].tile>=0)
+                int tx = i % w;
+                int ty = i / w;
+                const auto& cf = cur_frame[i];
+                if(cf.tile>=0)
                 {
-                    int tx=i % w;
-                    int ty=i / w;
-                    db->tile[tx][ty]=cur_frame[i].tile;
-                    db->back[tx][ty]=cur_frame[i].back;
-                    db->bright[tx][ty]=cur_frame[i].bright;
-                    db->fore[tx][ty]=cur_frame[i].fore;
+                    db->tile[tx][ty]= cf.tile;
+                    db->back[tx][ty]= cf.back;
+                    db->bright[tx][ty]= cf.bright;
+                    db->fore[tx][ty]= cf.fore;
                 }
+                if (cf.graphics_tile >= 0)
+                    db->texpos1[tx][ty] = cf.graphics_tile;
+                if (cf.overlay_tile >= 0)
+                    db->texpos2[tx][ty] = cf.overlay_tile;
+                if (cf.unk_tile >= 0)
+                    db->texpos3[tx][ty] = cf.unk_tile;
```

Comment:
let me update the structures so these are named the same as the fields in `enabler`

edit: https://github.com/DFHack/df-structures/pull/737